### DISCUSSION TITLE
keyboard shortcuts for buttons

### DISCUSSION THREAD

**Discussion by tfriedel** [marimo-team/marimo#612] [Ideas]
I quite liked the example for data labeling. 
There are buttons to go to the next or previous image. It would be nice to attach keyboard shortcuts to buttons so such a tool could be used without a mouse.

**Comment by mscolnick** [comment]
This is a good feature request - I will create an issue for this. To see if anyone wants to pick it up.

**Comment by rithik56** [comment]
@tfriedel @mscolnick can you please elaborate on the issue in more detail since I am new to this project.

**Comment by tfriedel** [comment]
@rithik56 For the use case of a data labeling tool, it's helpful to control actions like assigning a label or skipping to the next or previous item with a keyboard shortcut. In some UI frameworks you can assign hotkeys to buttons, e.g. you could have a button labeled "Next" that is activated by clicking on it or pressing the "right arrow" key. It may also be useful to add the feature to other widgets like Sliders. 
Or even have a non-widget specific keyboard press listener.

It may make sense to check out how this feature is possible in other frameworks.
E.g. in panel:
https://github.com/holoviz/panel/issues/3193

streamlit:
https://github.com/adriangalilea/streamlit-shortcuts

react:
https://devtrium.com/posts/how-keyboard-shortcut

**Comment by rithik56** [comment]
@tfriedel @mscolnick thanks for clarifying the doubt but I am having a hard time understanding how to add this feature ?
